From 848cac7e6099e342df57ed9778f8b1406d856301 Mon Sep 17 00:00:00 2001
From: karanx <karanx.bhagoji@intel.com>
Date: Tue, 20 Feb 2018 20:07:27 +0530
Subject: [PATCH 05/11] usb: dwc3: usb limit to high speed.

This works around a 'adb root' and cable disconnect-connect issue when
USB OTG is connected to a superspeed port.

o This patch limits USB max speed on a superspeed port to high speed.
o A USB-C to USB-C cable will still experience a disconnect-reconnect
issue. Use a USB-A to USB-C cable.

This patch needs to be added for the AIA-O for the NUC platform
for USB3.0 proper detection.

Jira: None.
Test: Depends on patch W/A for dwc3 revision < 2.2a
      Boot the board. Check adb with USB3.0 cable.

BUG: 32940866
Change-Id: I00a888946d729078728768a5b3327abf2fda81cc
(cherry picked from commit 483946c1afce6c714d15d4ea981fdefee86fb555)
Signed-off-by: mabbas <mohamed.abbas@intel.com>
Signed-off-by: Abhilash K V <abhilash.k.v@intel.com>
Signed-off-by: karan889 <karanx.bhagoji@intel.com>
---
 drivers/usb/dwc3/core.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/dwc3/core.c b/drivers/usb/dwc3/core.c
index dca78bb..c33d973 100644
--- a/drivers/usb/dwc3/core.c
+++ b/drivers/usb/dwc3/core.c
@@ -1028,6 +1028,9 @@ static void dwc3_get_properties(struct dwc3 *dwc)
 	hird_threshold = 12;
 
 	dwc->maximum_speed = usb_get_maximum_speed(dev);
+	if (dwc->maximum_speed > USB_SPEED_HIGH)
+		dwc->maximum_speed = USB_SPEED_HIGH;
+
 	dwc->dr_mode = usb_get_dr_mode(dev);
 	dwc->hsphy_mode = of_usb_get_phy_mode(dev->of_node);
 
@@ -1142,7 +1145,7 @@ static void dwc3_check_params(struct dwc3 *dwc)
 		/* fall through */
 	case USB_SPEED_UNKNOWN:
 		/* default to superspeed */
-		dwc->maximum_speed = USB_SPEED_SUPER;
+		dwc->maximum_speed = USB_SPEED_HIGH;
 
 		/*
 		 * default to superspeed plus if we are capable.
-- 
1.9.1

